math-scala/src/main/scala/org/apache/mahout/math/drm/RLikeDrmOps.scala

```diff
@@ -50,7 +50,11 @@ class RLikeDrmOps[K: ClassTag](drm: DrmLike[K]) extends DrmLikeOps[K](drm) {
 
   def *:(that: Double): DrmLike[K] = OpAewUnaryFunc[K](A = this, f = that * _)
 
-  def ^(that: Double): DrmLike[K] = OpAewUnaryFunc[K](A = this, f = math.pow(_, that))
+  def ^(that: Double): DrmLike[K] = that match {
+    case 2.0 ⇒ OpAewUnaryFunc[K](A = this, f = x ⇒ x * x)
```

Comment:
Can we have a comment here explaining the reason behind special-case for 2.0 and 0.5? This will help future code readers. Same comment applies to rest of the similar changes further below.